salt/cloud/clouds/azurearm.py

```diff
@@ -260,6 +263,14 @@ def __is_provider_configured(opts, provider, required_keys=()):
             ("subscription_id", "username", "password"),
         )
 
+    if provider is False:
+        # check if using MSI style credentials...
+        provider = config.is_provider_configured(
+            __opts__,
+            __active_provider_name__ or __virtualname__,
+            required_keys=("subscription_id",),
+        )
+
```

Comment:
From your question in Slack - we should probably refactor this code. But at minimum, we need to test:

1. if provider is found by `("subscription_id", "tenant", "client_id", "secret")`, that provider is returned
2. if provider is *not* found that way, but is found via `("subscription_id", "username", "password")`, it is returned
3. if provider is not found either of those ways, then whatever is returned from `config.is_provider_configured(...)` is returned.

Presumably that's False, but it would be more consistent and expected for these things to return None than False.

Reply:
@nicholasmhughes Excellent job! I think we do still need the other test to be able to merge this, but that should be much more straightforward? 😂